pypit/fluxspec.py

```diff
@@ -226,44 +233,54 @@ def generate_sensfunc(self):
         self.sensfunc : dict
 
         """
+        # Check internals
         if self.std is None:
-            msgs.warn("You need to identify the Star first (with find_standard)") #load in the Standard spectra first")
+            msgs.warn('First identify the star first (with find_standard).')
             return None
         if self.std_header is None:
-            msgs.warn("You need to set the std_header with a dict-like object holding RA, DEC, AIRMASS, EXPTIME")
+            msgs.warn('First set std_header with a dict-like object holding RA, DEC, '
+                      'AIRMASS, EXPTIME.')
             return None
-        # Run
+
+        # Get extinction correction
+        extinction_corr = arflux.extinction_correction(self.std.boxcar['wave'],
+                                                       self.std_header['AIRMASS'],
+                                                       self.extinction_data)
         self.sensfunc = arflux.generate_sensfunc(self.std, self.std_header['RA'],
-                                            self.std_header['DEC'], self.std_header['AIRMASS'],
-                                            self.std_header['EXPTIME'], self.settings)
+                                                 self.std_header['DEC'],
+                                                 self.std_header['EXPTIME'], extinction_corr)
+
         # Step
         self.steps.append(inspect.stack()[0][3])
         # Return
         return self.sensfunc
 
-    def _flux_specobjs(self, specobjs, airmass, exptime):
-        """
-        Flux an input list of SpecObj objects
-          Can be packed in detector, slit, etc. or as a simple list
-
-        Wrapper to arflux.apply_sensfunc()
-
-        Parameters
-        ----------
-        specobjs : list
-        airmass : float
-        exptime : float
-
-        Returns
-        -------
-
-        """
-        # Note the unravel here
-        for sci_obj in arutils.unravel_specobjs(specobjs):
-            if sci_obj is not None:
-                # Do it
-                arflux.apply_sensfunc(sci_obj, self.sensfunc, airmass, exptime, self.settings)
-
+# Does not seem to be used
```

Comment:
Might be called by the fluxing script.  Please check.

Reply:
This is definitely used by arflux.py

Reply:
I don't see it in my current branch.  But I've uncommented this function and removed the leading underscore.  Do we need both `flux_specobjs` and `flux_science`?